src/App.js

```diff
@@ -7,29 +7,39 @@ import axios from 'axios';
 
 class App extends React.Component {
 
-  constructor(props){
+  constructor(props) {
     super(props);
     this.state = [
-      {name: "Breakfast"},
-      {name: "Lunch"},
-      {name: "Dinner"},
-      {name: "Sides"},
-      {name: "Dessert"}
+      { name: "Breakfast", Menu: "" },
+      { name: "Lunch", Menu: "" },
+      { name: "Dinner", Menu: "" },
+      { name: "Sides", Menu: "" },
+      { name: "Dessert", Menu: "" }
     ]
   }
-  
-   componentDidMount(){
-
-     
-     this.state.map(MenuItems => {
-       axios.get('https://entree-f18.herokuapp.com/v1/menu/12')
-       .then (items => {
-         localStorage.setItem(MenuItems.name, JSON.stringify(items.data.menu_items))
-       })
-      ;
-     })
-     
-   
+
+  componentDidMount() {
+
+    // Check if menu data already exists in local storage.
+    //If so, store data in local storage in menu state, else pull from API
+
+    if (localStorage.getItem('Breakfast')) {
+      this.state.map(MenuItems_Local => {
+        this.state.Menu = JSON.parse(localStorage.getItem(MenuItems_Local.name));
```

Comment:
You should use `this.setState` whenever you're changing a value in your component state. I also don't think `this.state.Menu` is the value you want to change here.

Reply:
`MenuItems_Local.Menu`

Reply:
This is very helpful to know.  I am actually removing the Menu objects from the state object and will be referencing the local storage directly depending on the page.